pennylane/ops/qubit/non_parametric_ops.py

```diff
@@ -1018,46 +1192,48 @@ def __init__(
         ):
             raise ValueError("The work wires must be different from the control and target wires")
 
-        self._target_wire = wires[0]
-        self._work_wires = work_wires
-        self._control_wires = control_wires
-
-        wires = control_wires + wires
-
         if not control_values:
             control_values = "1" * len(control_wires)
 
-        control_int = self._parse_control_values(control_wires, control_values)
-        self.control_values = control_values
+        self._hyperparameters = {
+            "target_wire": wires[0],
+            "work_wires": work_wires,
+            "control_wires": control_wires,
+            "control_values": control_values,
+        }
```

Comment:
Same changes as in `ControlledQubitUnitary`: We save the hyperparameters and then compute matrices in the matrix methods...